### PR TITLE
chore(config): cleanup app-config files

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

## Which issue(s) does this PR fix

- Unblocks https://github.com/janus-idp/backstage-showcase/pull/602

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

**Comment by schultzp2020** [comment]
@PatAKnight can you verify that this does not impact you?

**Review by PatAKnight — APPROVED** [review]
LGTM

File: app-config.yaml [hunk 1]
```diff
@@ -21,8 +21,9 @@ app:
         headerColor1: ${HEADER_DARK_COLOR_1}
         headerColor2: ${HEADER_DARK_COLOR_2}
         navigationIndicatorColor: ${NAV_INDICATOR_DARK_COLOR}
+
 organization:
-  name: My Org
+  name: ${ORGANIZATION_NAME}
```
Comment (kadel) [inline app-config.yaml:26]: It looks like this introduces new variable, we should document it in https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md

File: app-config.production.yaml [hunk 2]
```diff
@@ -33,12 +34,3 @@ backend:
       ssl:
         ca: # if you have a CA file and want to verify it you can uncomment this section
           $file: /mnt/certs/ca.crt
-
-auth:
-  environment: production
```
Comment (kadel) [inline app-config.production.yaml:38]: Is it possible to keep just following? 
```yaml

auth:
  environment: production
```

**Review by kadel — CHANGES_REQUESTED** [review]
There is a lot of changes  and it invalidates most of the things in https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md it should be updated to reflect this change

**Comment by schultzp2020** [comment]
@kadel I don't think it's worth while to update the getting started docs because it will need a huge rewrite because of dynamic plugins. We were also debating on deleting the non example `app-config`s.

**Comment by kadel** [comment]
> @kadel I don't think it's worth while to update the getting started docs because it will need a huge rewrite because of dynamic plugins.

We should try to keep it up to date where it makes sense. If we let updates pile up it something important could get lost.



> We were also debating on deleting the non example `app-config`s.

+1 to that, we need to change the approach to configs and reduce the number of overlayed files.

**Comment by davidfestal** [comment]
I agree that such a change will be required in a near future.
But it seems to me that it might be premature, since the [new Helm chart PR](https://github.com/janus-idp/helm-backstage/pull/111) is not even merged, and the frontend dyamic plugins is still in the process of landing, but not fully merged in upstream.

Until the switch to dynamic plugins is completely done (frontend + backend + helm chart), I think it is not a good choice to break compatibility with how the `enable.xxx` properties and default config file values used to work.  

To me the plan should be the following:
- finish frontend dynamic plugins integration
- merge / publish the new Helm Chart
- then the new helm chart will provide app-config values which will override those in the app-config.xxx.yaml files included in the container image. So this will work, while keeping compatibility with the old Helm chart.
- When the new Helm chart is published and widely used, and we have agreed on and documented the fact that no plugin is loaded by default, then we would be able to cleanup the app-config.xxx.yaml files the way it is done here, without breaking any user expectation.

**Review by davidfestal — CHANGES_REQUESTED** [review]
Seems premature to me (see comment [here](https://github.com/janus-idp/backstage-showcase/pull/694#issuecomment-1790644823)), though I'm not saying that part of this cleanup could not be merged now.

**Comment by christophe-f** [comment]
This is a blocking PR. So we should keep it to the minimum to unblock the conditional login. There is a clean up needed but that can be done later as it seems that there is a lot of considerations to look at.

**Comment by schultzp2020** [comment]
@davidfestal the non example `app-config`s are actually our configs. The only config the user worries about is the example config because they will supply their own config.

**Comment by schultzp2020** [comment]
> Until the switch to dynamic plugins is completely done (frontend + backend + helm chart), I think it is not a good choice to break compatibility with how the enable.xxx properties and default config file values used to work.

The compatibility will not be broken because the `enabled` section of the config will be added by the respective dynamic plugin installation.

**Comment by schultzp2020** [comment]
@davidfestal and I just had a quick call about this. We think the best way forward is to create a new `app-config` called `app-config.deprecated.yaml` that is a copy of the old `app-config.yaml`. This way users can still get the old behavior by adding an arg in the helm chart to point to the old `app-config` at `app-config.deprecated.yaml`. Thoughts? @christophe-f @kadel

**Comment by tumido** [comment]
Does showcase have a version associated with it? We can always say the Helm chart supports images up to certain `appVersion`:

https://helm.sh/docs/topics/charts/#the-appversion-field

I think the `app-config.deprecated.yaml` is a good solution if we need to offer backward compatibility. The question is - do we need to offer that?

**Comment by kadel** [comment]
> Does showcase have a version associated with it? We can always say the Helm chart supports images up to certain `appVersion`:
> 
> https://helm.sh/docs/topics/charts/#the-appversion-field
> 
> I think the `app-config.deprecated.yaml` is a good solution if we need to offer backward compatibility. The question is - do we need to offer that?

I wouldn't worry about backward compatibility until we are at 1.0

**Review by kadel — APPROVED** [review]

**Review by davidfestal — APPROVED** [review]
Approving, to let others decide.

**Pull request merged**